drivers/timer/xlnx_psttc_timer.c

```diff
@@ -3,10 +3,12 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <kernel.h>
 #include <drivers/timer/system_timer.h>
-#include <sys_clock.h>
-#include "irq.h"
-#include "legacy_api.h"
+
+#ifdef CONFIG_TICKLESS_KERNEL
+#error "Tickless mode is not supported"
+#endif
```

Comment:
FWIW: you could just drop this patch and squash it with the later ones, though it's not a big deal.  The only meat is this error and the z_clock_elapsed().  The latter actually has a weak equivalent that gets linked in if a driver doesn't have its own, so this is actually a noop.

As far as the error: the way it works is that "TICKLESS_KERNEL" really means that the kernel will present the appropriate interface to the timer driver (i.e. asking for specific timeouts via z_clock_set_timeout()) to allow it to operate in a tickless manner.  Whether it does or not is... essentially the driver's business.  It's still legal for a driver to ignore those calls and just issue a z_clock_announce() with an argument of 1 for every tick.

Basically this error isn't wrong, but it's IMHO needlessly pedantic.  In fact building a tickless kernel with this non-tickless driver will work just fine.  It just won't be tickless.

Reply:
Thanks for reviewing.

> FWIW: you could just drop this patch and squash it with the later ones

I wanted to keep an intermediate stage before completely moving to the "Match" mode, as there might be an unforeseen issue with it and that commit might need to be reverted in the future, in which case the refactoring done in this commit won't be lost.

> It's still legal for a driver to ignore those calls and just issue a z_clock_announce() with an argument of 1 for every tick. ... Basically this error isn't wrong, but it's IMHO needlessly pedantic. 

In that case, error does sound to be a little too pedantic: I will replace the error with a warning then.